Userland/Libraries/LibELF/DynamicObject.cpp

```diff
@@ -507,6 +507,11 @@ Optional<DynamicObject::SymbolLookupResult> DynamicObject::lookup_symbol(const E
     if (symbol.is_undefined() || symbol.bind() == STB_WEAK)
         return DynamicLinker::lookup_global_symbol(symbol.name());
 
+    if (symbol.bind() == STB_GLOBAL) {
+        if (auto res = DynamicLinker::lookup_global_symbol(symbol.name()); res.has_value())
+            return res;
+    }
```

Comment:
@ADKaster @itamar8910 I don't know if this is right (what I've read so far suggests that it could be), but it does resolve some issues I encountered.

In general, this commit is probably weird, but I think it's better to have a half-working implementation and fix it later - please tell me if FIXME's are needed in here somewhere

Reply:
Oops, I made all those comments on your commit and not the PR 🤦‍♂️. Phone GitHub strikes again!

Anyway re RTLD_NEXT and RTLD_DEFAULT, they're actually reserved by POSIX with the expected behavior defined in the Application Usage section of dlsym. https://pubs.opengroup.org/onlinepubs/009696699/functions/dlsym.html

I think they're pretty useful for applications that load plugins for finding "the symbol foo as if there was a relocation for it" and for things like LibC replacements where you want to LD_PRELOAD a custom malloc or to override a symbol provided in another shared object when using it in this one.

Reply:
Alright, I'll add a FIXME about it, why not.

Reply:
(for future reference - mostly for myself lol - the comments are here: https://github.com/SerenityOS/serenity/commit/c0db5db28a829531be480b7f6dda5c067c03f440)